core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -949,6 +949,85 @@ public void modeInUnsupportedFormat() {
         exception.getMessage());
   }
 
+  private void testWeek_Of_Year(String date, int mode, int expectedResult) {
+    FunctionExpression expression = DSL
+        .week_of_year(DSL.literal(new ExprDateValue(date)), DSL.literal(mode));
+    assertEquals(INTEGER, expression.type());
+    assertEquals(String.format("week_of_year(DATE '%s', %d)", date, mode), expression.toString());
+    assertEquals(integerValue(expectedResult), eval(expression));
+  }
+
+  private void testNullMissingWeek_Of_Year(ExprCoreType date) {
+    when(nullRef.type()).thenReturn(date);
+    when(missingRef.type()).thenReturn(date);
+    assertEquals(nullValue(), eval(DSL.week_of_year(nullRef)));
+    assertEquals(missingValue(), eval(DSL.week_of_year(missingRef)));
+  }
+
+  @Test
+  public void week_of_year() {
```

Comment:
Split this up into different tests to help debugging in the event of a failure.

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/175/commits/b6c94ae6ceae47411508deb3f0c8a461fef4f700 and https://github.com/Bit-Quill/opensearch-project-sql/pull/175/commits/e937045d647b7834674144cbdf8ea4828eff314d